pylint/checkers/base/basic_error_checker.py

```diff
@@ -285,14 +285,17 @@ def visit_functiondef(self, node: nodes.FunctionDef) -> None:
 
         # provide detailed report about each repeated argument
         for argument_duplicates in arg_clusters.values():
```

Comment:
@Pierre-Sassoulas Can't we just merge this with for loop above? Add `arg.name` to a list. If `name` is already in that list raise a message on the current `arg` node being checked?

Reply:
Yeah thank you, I was in the mindset of extending the existing code, but a refactor is a lot better.